crate_universe/src/rendering/template_engine.rs

```diff
@@ -398,3 +404,79 @@ fn crate_repository_fn_generator(template: String, repository_name: String) -> i
         },
     )
 }
+
+/// Tera filter which re-keys a [`SelectList`]. See [`SelectList::remap_configurations`]
+/// for more details. The mapping must be provided as a Tera argument named `mapping`.
+fn remap_select_list_configurations<T: Ord + Clone + for<'a> Deserialize<'a> + Serialize>(
+    input: &Value,
+    args: &HashMap<String, Value>,
+) -> tera::Result<Value> {
+    let mapping = parse_tera_param!("mapping", BTreeMap<String, BTreeSet<String>>, args);
+
+    match from_value::<SelectList<T>>(input.clone()) {
+        Ok(v) => match to_value(v.remap_configurations(&mapping)) {
+            Ok(v) => Ok(v),
+            Err(_) => Err(tera::Error::msg("Failed to remap conditions.")),
+        },
+        Err(_) => Err(tera::Error::msg("The filter input could not be parsed.")),
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::config::Config;
+    use crate::metadata::Annotations;
+    use crate::test::*;
+
+    /// Trims leading and trailing whitespace from each line.
+    fn trim_lines(input: &str) -> String {
+        input.lines().map(str::trim).collect::<Vec<_>>().join("\n")
+    }
+
+    #[test]
+    fn test_render_build_file_deps() {
```

Comment:
Could this test be moved to `rendering.rs`? It doesn't seem like it's specific to Tera (though that's not really a strong distinction since there's only one template engine 😅)

Reply:
Yup... done. That's definitely nicer.